### ISSUE TITLE
New Ubuntu based VMs gets same IP

### ISSUE THREAD

**Issue by ranjujohn** [kubermatic/machine-controller#1455]
We came across an issue that all new Ubuntu based VMs gets the same IP in a vSphere environment. This would have happened due to the fact that someone would have started the template VM and the machine-id would have generated. 

Will it be possible to achieve one of the below options to overcome this issue?

- Machine-controller checks if `machine-id `is set in the template VM and remove it as part of cloning.
- Convert the VM to a template and use that? So that we can avoid accidental starting of the template VM?
- Configure netplan to use `mac` as the `dhcp-identifier`

**Comment by sphr2k** [comment]
IMO, this is not machine-controller's job, but something that should be taken care of when creating the base image, i.e. clear the machine-id and optionally change netplan config as you mentioned. 

Alternatively, I think you should be able to take care of clearing the machine-id using [operating-system-manager](https://github.com/kubermatic/operating-system-manager/).

**Comment by toschneck** [comment]
yes agree should be done with OSM. @csengerszabo could move the issue?
